rpcutils/rpcclient.go

```diff
@@ -448,6 +449,53 @@ func CommonAncestor(client *rpcclient.Client, hashA, hashB chainhash.Hash) (*cha
 	return &hashA, chainA, chainB, nil
 }
 
+// OrphanedTipLength finds a common ancestor by iterating block heights
+// backwards until a common block hash is found. Unlike CommonAncestor, an
+// orphaned DB tip whose corresponding block is not known to dcrd will not cause
+// cause an error. The number of blocks that have been orphaned is returned.
+// Realistically, this should rarely be anything but 0 or 1, but no limits are
+// placed here on the number of blocks checked.
+func OrphanedTipLength(ctx context.Context, client *rpcclient.Client,
```

Comment:
Trying to think how this could bet tested via go test.  One idea is to replace `*rpcclient.Client` with `BlockHashGetter` so you can stub out the client.  I think it would be worth testing all the edge cases by manipulating what the `BlockHashGetter` and the `hashFunc` return.

Reply:
Thanks for the hint here. I already had the test written, but with a `rpcclient.Client` with my local dcrd credentials, so I didn't include it.